assets/translate.js

```diff
@@ -124,71 +124,80 @@ $( function () {
  * When a translation field is changed, update the image preview, and also mark the form as unsaved.
  */
 $( window ).on( 'load', function () {
-	$( '.translation-fields .oo-ui-fieldLayout .oo-ui-inputWidget' ).each( function () {
-		var inputWiget = OO.ui.infuse( $( this ) ),
-			$imgElement = $( '#translation-image img' ),
-			targetLang = $( ':input[name="target-lang"]' ).val(),
-			updatePreviewImage = function () {
-				var requestParams = {},
-					canUpload = false,
-					$uploadButtonElement = $( '#upload-button-widget' ),
-					uploadButtonWidget = OO.ui.infuse( $uploadButtonElement );
-
-				if ( window.alreadyUpdating ) {
-					// Otherwise, it will needlessly update when the input is being enabled
-					return;
-				}
-				window.alreadyUpdating = true;
-				// Show loading indicator.
-				$( '.image-column' ).addClass( 'loading' );
-				// Go through all fields and construct the request parameters.
-				$( '.translation-fields .oo-ui-fieldLayout' ).each( function () {
-					var textChanged,
-						originalText = '',
-						fieldLayout = OO.ui.infuse( $( this ) ),
-						tspanId = fieldLayout.getField().data[ 'tspan-id' ],
-						text = fieldLayout.getField().getValue();
-					if ( appConfig.translations[ tspanId ][ targetLang ] !== undefined ) {
-						originalText = appConfig.translations[ tspanId ][ targetLang ].text;
+	$( function () {
```

Comment:
I'm sort of confused as to why we need to wait for the 'ready' event here when we're in a handler for the 'load' event which (I thought) happens after the former. But this is just a drop in the ocean of things that I don't really get about jquery and Javascript, so I'm happy to just confirm it works.

Reply:
I am confused too! Your understanding is the same as mine. For some reason the rules didn't seem to apply here... I only got it to work with `load` followed by `ready`.